src/main/java/org/spongepowered/api/map/util/MapPaletteFactory.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.map.util;
+
+import com.google.common.collect.ImmutableCollection;
+import org.spongepowered.api.map.color.MapPalette;
+import org.spongepowered.api.map.color.MapColor;
+import org.spongepowered.api.map.color.MapShade;
+import org.spongepowered.api.util.Color;
+
+/**
+ * Represents the required implementation for the static methods in
+ * {@link MapPalette}.
+ */
+public interface MapPaletteFactory {
```

Comment:
So that the methods on MapPalette are static?

On Sat, Apr 2, 2016, 2:49 PM Aaron Hill <notifications@github.com> wrote:

> In src/main/java/org/spongepowered/api/map/util/MapPaletteFactory.java
> <https://github.com/SpongePowered/SpongeAPI/pull/1162#discussion_r58298677>
> :
>
> > + * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
> > + * THE SOFTWARE.
> > + */
> > +package org.spongepowered.api.map.util;
> > +
> > +import com.google.common.collect.ImmutableCollection;
> > +import org.spongepowered.api.map.color.MapPalette;
> > +import org.spongepowered.api.map.color.MapColor;
> > +import org.spongepowered.api.map.color.MapShade;
> > +import org.spongepowered.api.util.Color;
> > +
> > +/**
> > + * Represents the required implementation for the static methods in
> > + * {@link MapPalette}.
> > + */
> > +public interface MapPaletteFactory {
>
> What's the point of this?
>
> —
> You are receiving this because you authored the thread.
>
> Reply to this email directly or view it on GitHub
> <https://github.com/SpongePowered/SpongeAPI/pull/1162/files/d8338269be828e278d15ed5ffbe9bb30b7c899c1#r58298677>
>